### PR TITLE
NodeAdmin: Remove usage for Environment

### PR DESCRIPTION
**Pull request by freva**
This is the second last (planned) PR of cleanup in `node-admin` (the last one will remove several classes/files that are no longer used).

Removes usage of `Environment` by:
* Adding `DockerNetworking` and `ZoneId` to `NodeAgentContext`
* Moves spec verification as a task in `host-admin`
* Moves filebeat config generation to `host-admin`, added in `ContainerData`
* Makes `AclMaintainer` optional (to be disable-able for docker host networking)

### CODE REVIEW

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeadmin/NodeAdminStateUpdaterImpl.java [hunk 1]
```diff
@@ -54,55 +45,34 @@
...
         this.nodeAdmin = nodeAdmin;
-        this.dockerHostHostName = dockerHostHostName;
+        this.hostHostname = hostHostname.value();
         this.clock = clock;
         this.nodeAdminConvergeStateInterval = nodeAdminConvergeStateInterval;
-        this.classLocking = classLocking;
         this.lastTick = clock.instant();
 
         this.loopThread = new Thread(() -> {
-            if (classLocking.isPresent()) {
```
Comment (hakonhall) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeadmin/NodeAdminStateUpdaterImpl.java]: Great!

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeagent/NodeAgentContext.java [hunk 2]
```diff
@@ -19,6 +21,10 @@
 
     AthenzService identity();
 
+    DockerNetworking dockerNetworking();
+
+    ZoneId zone();
```
Comment (hakonhall) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeagent/NodeAgentContext.java]: zoneId()

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeagent/NodeAgentContextImpl.java [hunk 3]
```diff
@@ -97,4 +119,66 @@ public void log(Logger logger, Level level, String message) {
     public void log(Logger logger, Level level, String message, Throwable throwable) {
         logger.log(level, logPrefix + message, throwable);
     }
+    
+
+    /** For testing only! */
+    public static class Builder {
```
Comment (hakonhall) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeagent/NodeAgentContextImpl.java]: Does it have to be public?

Comment (freva) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeagent/NodeAgentContextImpl.java]: Yes, this is used in som `vespa/hosted` tests as well

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeagent/NodeAgentContext.java (hunk 2 above)
Comment (freva) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeagent/NodeAgentContext.java]: Fixed.

**Review by hakonhall — APPROVED** [review]

**Pull request merged**